### ISSUE TITLE
Crash (Signal 11) when trying to process an unsorted BAM

### ISSUE THREAD

**Issue by maplesond** [jwalabroad/VariantBam#2]
A crash (Signal 11) occurs when trying to process an unsorted BAM.  The software could check the SAM header to see if the file is file is not-sorted up before processing and provide a useful error message if this is the case.

**Comment by walaj** [comment]
Hi Daniel, 
Could you provide the command that you used to get this error? I can't re-create, so it could be specific to a certain filter set.

**Comment by maplesond** [comment]
Here's the command line: "variant bowtie2.bam -m 100 -o b2_mc100.bam"

It worked fine after I sorted bowtie2.bam with samtools.

**Comment by walaj** [comment]
OK I see the issue, it is segfaulting when trying to downsample to a certain coverage, which for that filter expects a sorted BAM. I'll fix to have it give a more informative error before I close this thread.

Edit: I added a heuristic check for a sorted BAM, that will cause it to give a useful error message if it sees an unsorted BAM when trying to downsample. Updated the documentation to indicate that sorted BAMs are required for these filters.
